sympy/core/add.py

```diff
@@ -70,6 +70,48 @@ def _unevaluated_Add(*args):
 
 
 class Add(Expr, AssocOp):
+    """
+    Addition operation for algebraic group.
```

Comment:
I didn't realise that `Add` doesn't even have a docstring.

I would suggest that these docs should be focussed on end users. A user probably doesn't want to know about `MatAdd` etc but if we could link to some information about kinds then that would be good. There could be a "see also" linking to more information like `MatAdd` or some page in the docs that explains the general relationship between different `Add` classes.

We should first make it clear that this is how an expression that is a simple sum is represented and that `Add(x, y)` is equivalent to `x + y`. Showing `evaluate=False` is good.

It would be good to document the basic algorithm that is used in `Add` when it evaluates:

1. Single arg `Add(x) -> x` (even if `evaluate=False`)
2. Flatten using associativity: `Add(x, Add(y, z)) -> Add(x, y, z)`.
3. Split each term into a `Number` coefficient and a "term" using `as_coeff_Mul`.
4. Collect `Number` coefficients for each "term" and add them using `+`.
5. Remove terms with zero coefficient
6. Sort the terms and insert pure number at `args[0]`
7. Also some special handling for things like infinity.

Hence:
```
In [73]: 2*x**2 + 3*x + I*y + 2*y + 2*x/5 + 1.0*y + 1
Out[73]: 
   2   17⋅x              
2⋅x  + ──── + 3.0⋅y + ⅈ⋅y
        5  
```
It's worth noting that the printers don't display an Add in args order:
```
In [80]: Add(x, 1)
Out[80]: x + 1

In [81]: Add(x, 1).args
Out[81]: (1, x)
```

Another thing to mention could be that with many terms `Add(*args)` is more efficient than `sum(args)` or anything else based on repeated addition since the algorithm used for evaluating an `Add` is linear in the number of terms.

Reply:
Thinking about this the most  likely way that a user would see this docstring is by doing `help(obj)` without having any idea that `obj` is an `Add`. The docstring should probably contextualise itself from the perspective that this is an `Expr` representing an expression of some kind.

Reply:
I updated the docstring.